### PR TITLE
Unit Tests: Introduced separate test suites.

### PR DESCRIPTION
**Pull request by zinigor**
This PR introduces test suites that will let us run phpunit over smaller portions of code at a time rather than running the whole deal every time. You can try it out by running `phpunit` with the `testsuite` switch:

```
$ phpunit --testsuite photon
```
The following test suites are made available by this PR:

- general
- opengraph
- photon
- heartbeat
- json-api
- media
- postimages
- contact-form
- infinity-scroll
- sharedaddy
- shortcodes
- atd
- widgets

### CODE REVIEW

**Comment by kraftbj** [comment]
Should we match the module slugs, e.g. `after-the-deadline`, when applicable?

File: phpunit.xml.dist [hunk 1]
```diff
@@ -1,8 +1,49 @@
...
+			<directory prefix="test_" suffix=".php">tests/php/modules/infinite-scroll</directory>
+		</testsuite>
+		<testsuite name="sharedaddy">
+			<directory prefix="test_" suffix=".php">tests/php/modules/sharedaddy</directory>
+		</testsuite>
+		<testsuite name="shortcodes">
+			<directory prefix="test_" suffix=".php">tests/php/modules/shortcodes</directory>
+		</testsuite>
+		<testsuite name="atd">
+			<directory prefix="test_" suffix=".php">tests/php/modules/test_class.after_the_deadline.php</directory>
```
Comment (kraftbj) [inline phpunit.xml.dist]: This should be in the `<file>` namespace, I believe. Alternatively, we can move the file into a folder to match the structure of the rest.

**Comment by enejb** [comment]
This looks great! :heart:

**Comment by zinigor** [comment]
@enejb thanks for the hint! That's great too if you are only interested in one test case, but the others will run as well, `filter` will only hide test output from everything else. I guess sometimes it's beneficial to use both `filter` and `testsuite`.

@kraftbj thanks for spotting the mistakes, I have fixed them in this iteration.

**Comment by eliorivero** [comment]
It's good to have testsuites to run groups of possibly unrelated tests which might be more cumbersome to invoke manually using `--filter`.
lgtm :+1:

**Pull request merged**